trin-core/src/portalnet/overlay.rs

```diff
@@ -514,8 +516,17 @@ where
                     // Init uTP stream if `connection_id`is received
                     Content::ConnectionId(conn_id) => {
                         let conn_id = u16::from_be(conn_id);
-
-                        self.init_find_content_stream(enr, conn_id).await
+                        let content = self.init_find_content_stream(enr, conn_id).await?;
+                        let content_key = TContentKey::try_from(content_key)
+                            .expect("Error decoding valid content key.");
```

Comment:
Should we handle gracefully this exception?

Reply:
My thinking was that it is excessive, since the content key will already have been validated when the overlay network initiates the request [here](https://github.com/ethereum/trin/blob/master/trin-history/src/jsonrpc.rs#L165) - but I can add it I'm missing another case where an invalid content key might sneak through up to this point?

Reply:
On second thought, might as well add in the graceful handling, just in case.